apps/storefront/src/pages/ShoppingListDetails/components/ShoppingDetailFooter.tsx

```diff
@@ -334,17 +337,34 @@ function ShoppingDetailFooter(props: ShoppingDetailFooterProps) {
       }
 
       const lineItems = addLineItems(items);
-      const deleteCartObject = deleteCartData(items);
       const cartInfo = await getCart();
+
       if (allowJuniorPlaceOrder && cartInfo.data.site.cart) {
+        const deleteCartObject = deleteCartData(cartInfo.data.site.cart.entityId);
+
         await deleteCart(deleteCartObject);
-        await updateCart(cartInfo, lineItems);
+      }
+
+      const { success, warning, error } = await partialAddToCart(lineItems);
+
```

Comment:
I think we need the call to `b3TriggerCartNumber()` here if success from partial add

I noticed in the frontend path it fires immediately after items are added to cart with `createOrUpdateExistingCart`

We could trigger it whenever `success.length > 0` that way we match current behavior